aten/src/ATen/test/result_type_test.cpp

```diff
@@ -0,0 +1,296 @@
+#include "gtest/gtest.h"
+
+#include "ATen/ATen.h"
+#include "ATen/native/ResultType.h"
+
+using namespace at;
+
+template<typename T1, typename T2>
+void expectEqualElement(const T1& a, const T2& b, const char* test) = delete;
```

Comment:
"I can convert resultType tests to python but operand casting logic (which uses) is still only in C++"

I'm not sure I understand -- why can't you just expose the operand casting logic in python?  E.g. torch._result_type(tensors).

Reply:
On the hard to maintain, sorry I should have clarified.   Basically:

- We need python tests (because that's what most people use)
- Duplicating tests sucks
- The C++ API is not stable and changes frequently.

Thus, we've sort of just not tested C++ things at this level.